### ISSUE TITLE
Allocate label type for validation missions based on per-city need

### ISSUE THREAD

**Issue by misaugstad** [ProjectSidewalk/SidewalkWebpage#2530]
##### Brief description of problem/feature
I think that right now, we are just splitting the validation missions equally among the 4 major label types. Of course, we don't have an even distribution of labels among those label types, so there is an imbalance in which label types require more validations. Furthermore, this is different in different cities! In Seattle, 74% of the labels that don't have validations are curb ramp labels. But in SPGG, only a small handful of curb ramp labels need validations while 65% of the labels that need validations are obstacle labels!

##### Potential solution(s)
The best idea I have so far to address this would be to allocate those missions proportionally to the number of labels without validations of each label type. So if 75% of the labels without validations are curb ramps, then 75% of the validation missions will be curb ramp validation missions. Maybe once we are close to every label being validated (less than a thousand or so remaining) we switch to doing it proportionally based on just the number of labels of each label type.

I don't think that we want to do this query every time someone goes to the validation page. I figure we could run this query once per day or week and cache the numbers to be used whenever someone validates.

An alternative to this whole process might be to let the distribution be something that we can configure through the admin page. It would give us more fine-grained control over the distribution of missions, but it would be less automated.

**Comment by misaugstad** [comment]
>I don't think that we want to do this query every time someone goes to the validation page. I figure we could run this query once per day or week and cache the numbers to be used whenever someone validates.

I no longer agree with this piece. We've added a `correct` column to the `label` table, and we know that a label needs validations if `correct IS NULL`, so counting how many needs validations is actually really fast now! Shouldn't be an issue with doing this on every page load. Though I don't think that caching the data would be very difficult either :eyes:

**Comment by misaugstad** [comment]
We definitely need to do this! We are definitely wasting users' efforts when validating. I just did some testing in Pittsburgh this morning. I ran a query to check for the number of labels available for me to validate that have either no validations or disagree = disagree for each label type in Pittsburgh...
```
 label_type_id |   label_type   | count 
---------------+----------------+-------
             1 | CurbRamp       |  2166
             2 | NoCurbRamp     |    69
             3 | Obstacle       |  2137
             4 | SurfaceProblem |   408
             9 | Crosswalk      |    22
            10 | Signal         |    10
```

So there are tons of CurbRamp and Obstacle labels left for me to validate, some number of SurfaceProblem labels, and very few of the other label types remaining.

I then did a CurbRamp validation mission. The average number of previous validations for labels I validated was 0.4 validations; 6 of them had no validations yet, not bad!

For the Signal label type on the other hand, labels had an average of 4 previous validations, and none of them had zero validations!

For SurfaceProblem, labels had an average of 1.3 previous validations, with 4 of them having no previous validations (or agree = disagree).

It looks to me like our algorithm for choosing labels to validate is pretty good (it could be tuned slightly to see more labels that have no validations, but we do want some amount of redundancy). But the main issue we have is that some label types are just over-saturated with validations.
